packages/webamp-modern-2/src/skin/Clippath.ts

```diff
@@ -0,0 +1,167 @@
+export class Edges {
+  _top: string[];
+  _right: string[];
+  _bottom: string[] = [];
+  _left: string[] = [];
+
+  parseCanvasTransparency(canvas: HTMLCanvasElement) {
+    const w = canvas.width;
+    const h = canvas.height;
+    const ctx = canvas.getContext("2d");
+    const data = ctx.getImageData(0, 0, w, h).data;
+    let points: string[] = [];
+    var x, y, lastX, lastY;
+    var alpha: number;
+    var lastfoundx: number, lastfoundy: number, pending: boolean;
+    var first: boolean, found: boolean;
+
+    //? return true if not transparent
+    function fine(ax: number, ay: number): number {
```

Comment:
yeah, it was very confusing because initally I want to **crop**, but CSS said that she want to **keep**.
That way it named that for compromise 2 assumptions.
But now, maybe better to use "solid"? because it find anything that can keep the pixel in that surface being scanned.

Reply:
Ah. how about "opaque"?

Reply:
got it! I love that obvious meaning.